airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/ctid/InitialSyncCtidIterator.java

```diff
@@ -189,32 +208,65 @@ static List<Pair<Ctid, Ctid>> ctidQueryPlan(final Ctid startCtid,
                                               final int chunkSize,
                                               final double dataSize) {
     final List<Pair<Ctid, Ctid>> chunks = new ArrayList<>();
-    long lowerBound = startCtid.page;
-    long upperBound;
-    final double pages = dataSize / blockSize;
-    final long eachStep = (long) pages * chunkSize;
-    LOGGER.info("Will read {} pages to get {}GB", eachStep, chunkSize);
-    final long theoreticalLastPage = relationSize / blockSize;
-    LOGGER.debug("Theoretical last page {}", theoreticalLastPage);
-    upperBound = lowerBound + eachStep;
-
-    if (upperBound > theoreticalLastPage) {
-      chunks.add(Pair.of(startCtid, null));
-    } else {
-      chunks.add(Pair.of(Ctid.of(lowerBound, startCtid.tuple), Ctid.of(upperBound, 0)));
-      while (upperBound < theoreticalLastPage) {
-        lowerBound = upperBound;
-        upperBound += eachStep;
-        chunks.add(Pair.of(Ctid.of(lowerBound, 0), upperBound > theoreticalLastPage ? null : Ctid.of(upperBound, 0)));
+    if (blockSize > 0 && chunkSize > 0 && dataSize > 0) {
+      long lowerBound = startCtid.page;
+      long upperBound;
+      final double pages = dataSize / blockSize;
```

Comment:
is it possible that this calculation results in `pages` equal to 0? because in the next line `eachStep` will also be 0 and the while loop in line 227 will never end. In other words, do we need to check if pages is 0 and default to 1?

Reply:
The last test case I added in `InitialSyncCtidIteratorTest.testLegacyCtidQueryBounds()` is testing a table whose size fits a single page with 0 tuples in it (an empty table).
I modified that test slightly to have a single tuple.

Since pages is a double and `datasize` and `blockSize` are both `> 0`, how do you see this case happening

Reply:
purely on the calculation - if `blockSize` > `dataSize` then `pages` will be less than 0, and the cast to long will truncate it to 0 in line 215.

Reply:
See the test in `InitialSyncCtidIteratorTest#103`
Block size = 8K
Data size = 4K

You can make `eachStep` get to 0 if chunk size is small enough I guess

Reply:
ok, so if `eachStep` is 0 does the while loop in line 225 terminate? I realize this is new code, asking more so for my understanding

Reply:
No, you're right. 
I don't think it'll happen in read life but I'm making a change to protect from infinite loop just in case

Reply:
Updated both query plans